File: Integer/MergeSortedArray/MergeSortedArray.cc
class MergeSortedArray {
	public:
		void mergeSortedArray(int[] arrayA, int m, int [] arrayB, int n) {
			int index = m + n - 1;
			while (m > 0 && n > 0) {
				if(arrayA[m - 1] > arrayB[n - 1]) {
					arrayA[index] = arrayA[m - 1];
					--m;
				} else {
					arrayA[index] = arrayA[n - 1];
					--n;
				}
			}
			
			while(n > 0) {
				A[index] = B[n - 1];
				--n;
				--index;
			}
		}
};

File: Integer/MergeSortedArray/MergeSortedArray.py
class MergeSortedArray:
    def mergeSortedArray(self, arrayA, lenA, arrayB, lenB):
        if arrayB is none:
            return arrayA
            
        index = lenA + lenB - 1
        while lenA > 0 and lenB > 0:
            if arrayA[lanA - 1] > arrayB[lenB - 1]:
                arrayA[index] = arrayA[lenA - 1]
                lanA -= 1
            else:
                arrayA[index] = arrayB[lenB - 1]
                lenB -= 1
                
            index -= 1
            
        while lenB > 0:
            A[index] = B[lenB - 1]
            lenB -= 1
            index -= 1
        